pulsar-broker-common/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGeneratorUtils.java

```diff
@@ -111,5 +108,34 @@ static String getTypeStr(Collector.Type type) {
         }
     }
 
+
+    /**
+     * Write a label value to the writer, escaping backslashes, double quotes and newlines.
+     * See Promethues Exporter io.prometheus.client.exporter.common.TextFormat#writeEscapedLabelValue
+     */
+    public static String writeEscapedLabelValue(String s) {
+        if (s == null) {
+            return null;
+        }
```

Comment:
In order to reduce memory allocations we must search for bad characters initially and if the label is safe we should not create objects at all.